installMaGe.py

```diff
@@ -26,91 +18,128 @@
 python /path/to/installMaGe.py clean
 '''
 
-help_strings = ['h', '-h', '--h', 'help', '-help', '--help']
-# check for the right number of arguments
-if len(sys.argv) > 2 or (len(sys.argv) > 1 and sys.argv[1] in help_strings):
-    print(usage)
-    sys.exit()
+parser = argparse.ArgumentParser(description = usage)
+
+parser.add_argument('command', type=str,
+                    choices=['install', 'clean', 'reinstall'],
+                    help="What action to take. Options:\n"\
+                    "  install: install from scratch\n"\
+                    "  clean: uninstall and remove\n"\
+                    "  reinstall: clean and then install\n")
+parser.add_argument('-b', '--buildpath', type=str, default='.',
+                    help="Base path to build software in")
+parser.add_argument('-i', '--installpath', type=str,
+                    help="Base path to install software in. By default use the build path.")
```

Comment:
No strong option on this but alternatively you could consider installing in `~/.local` by default...

Reply:
That seems reasonable. @jasondet, @mpgreen, @oschulz, any opinions?